database/src/main/scala/cromwell/database/SqlDatabase.scala

```diff
@@ -178,22 +178,12 @@ trait SqlDatabase extends AutoCloseable {
                                                   (implicit ec: ExecutionContext):
   Future[Traversable[(Execution, ExecutionInfo)]]
 
-  protected def addMetadataEvent(workflowUuid: String,
-                                 key: String,
-                                 value: String,
-                                 valueType: String,
-                                 timestamp: Timestamp)
-                                (implicit ec: ExecutionContext): Future[Unit]
-
-  protected def addMetadataEvent(workflowUuid: String,
-                                 key: String,
-                                 callFqn: String,
-                                 index: Option[Int],
-                                 attempt: Int,
-                                 value: String,
-                                 valueType: String,
-                                 timestamp: Timestamp)
-                                (implicit ec: ExecutionContext): Future[Unit]
+  /**
+    * Add metadata events to the database transactionally. normalized type structure is as follows:
+    * (WorkflowId, MetadataKey, Option[CallFqn, CallIndex, CallAttempt], MetadataValue, MetadataValueType, Timestamp)
+    */
+  protected def addMetadata(events: Iterable[(String, String, Option[(String, Option[Int], Int)], String, String, Timestamp)])
+                           (implicit ec: ExecutionContext): Future[Unit]
```

Comment:
Good point, I think you're right.

At this point I'm pretty sour on the type aliases in general except for a couple of specific circumstances (mostly related to type wizardry from scalaz/cats). I'd rather stick with primitives for most situations if a case class can't be used.